### ISSUE TITLE
Handling MUC light messages with mod_event_pusher_http

### ISSUE THREAD

**Issue by cogentParadigm** [esl/MongooseIM#2217]
MongooseIM version: 3.2.0
Installed from: source
Erlang/OTP version: 21

Hi there. I have mod_event_pusher_http and mod_muc_light enabled and I implemented a callback module which sends groupchat messages but I'm only getting the message to the room itself at my http endpoint. Can you provide me any direction on how I could get it to handle the individual messages to participants?

So for example, given the following:

- user: `bob@local.com`
- user: `alice@local.com`
- room: `room@muclight.local.com`

If bob sends a message to `room@muclight.local.com`, then I will receive this POST:

```
{"author":"bob","server":"local.com","receiver":"room","message":"Bob's message"}
```

What I would like to receive is something more like this:

```
{"author":"bob","server":"local.com","receiver":"alice","message":"Bob's message"}
```

The important part is that this is the message going to alice. That way my callback_module can check if she is offline and my application server knows who needs to be notified.

Thanks!

**Comment by cogentParadigm** [comment]
I went on a bit of wild goose chase, but I managed to get this working.

At first I thought I had to implement additional hooks in `mod_event_pusher_hook_translator`. Initially, I tried `room_send_packet` but soon discovered that was invoked before the distribution to individual users so then I discovered `offline_groupchat_message_hook` which did catch the messages I wanted but they weren't getting through to my callback module's `should_make_req` function.

After some debugging I figured out that implementing the additional hook was unnecessary because `filter_local_packet` was already passing along those events. Where they get dropped is in `mod_event_pusher_http` in this function:

```erlang
push_event(Acc, _Host, #chat_event{direction = Dir, from = From, to = To, packet = Packet}) ->
    lists:map(fun(Opts) -> push_event(Acc, Dir, From, To, Packet, Opts) end,
              gen_mod:get_module_opt(From#jid.lserver, ?MODULE, configs, [])),
    Acc;
push_event(Acc, _Host, _Event) ->
    Acc.
```

The module configuration is loaded based on the host of the JID of the **sender**, which in my case would contain the `muclight.` prefix and therefore return an empty list.

For my use case, it's sufficient for me to change this to get the host from the JID of the **recipient** instead. Changing this line:

```erlang
gen_mod:get_module_opt(From#jid.lserver, ?MODULE, configs, [])),
```

To this:

```erlang
gen_mod:get_module_opt(To#jid.lserver, ?MODULE, configs, [])),
```

However, it doesn't seem like universally choosing one or the other is a good general purpose solution. If my understanding is correct, it's possible for either side to be external.

I'll leave it to you guys to determine if this issue should be closed or left open.
